<?xml version="1.0"?>

<launch>
    <!-- Set up hardware configurations and start up nodes related to hardware components. -->
    <group ns="hardware">
        <rosparam command="load" file="$(find expansion_board_driver)/config/hardware.yaml"/>
    </group>
    <node name="ybmotors_node" pkg="expansion_board_driver" type="ybmotors.py"/>
    <node name="ybfan_node"    pkg="expansion_board_driver" type="ybfan.py"/>
    <node name="ybleds_node"   pkg="expansion_board_driver" type="ybleds.py"/>

    <!-- Set up RPLiDAR-->
    <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode"/>
    <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
    <node name="map_to_laser_broadcaster"  pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1 map laser_frame 10"/>
</launch>